govcd/api.go

```diff
@@ -297,7 +297,10 @@ func checkRespWithErrType(resp *http.Response, err, errType error) (*http.Respon
 		return resp, nil
 	// Invalid request, parse the XML error returned and return it.
 	case
-		http.StatusBadRequest,                  // 400
+		http.StatusBadRequest,                   // 400
+		http.StatusRequestedRangeNotSatisfiable: // 416
+		return resp, fmt.Errorf(resp.Status)
```

Comment:
What's the reason for having errors 400 and 416 handled differently than the other 4* errors?
This change may alter the current API behavior

Reply:
See know issues in ticket https://github.com/vmware/go-vcloud-director/issues/328. For 400 and 416(as I know), it has no body in response, but our existing code is trying to decode body as error. So, got unexpected error:
Err: API Error: 0:
I already have a better fix for this after the current pull request. It will detect the error is "API Error: 0:" or not. If not, return directly. If true(means no body), return response status as error(response code + response message). So there is no specially handling for 400 416. Let me include it now.

Reply:
@dataclouder , Please help to review the new change for this.